stdlib/public/SDK/Dispatch/Time.swift

```diff
@@ -114,6 +114,30 @@ extension DispatchWallTime {
   }
 }
 
+
+// Returns m1 * m2, clamped to the range [Int64.min, Int64.max].
+// Because of the way this function is used, we can always assume
+// that m2 > 0.
+private func clampedInt64Product(_ m1: Int64, _ m2: Int64) -> Int64 {
+	assert(m2 > 0, "multiplier must be positive")
+	if m1 == 0 { return Int64(0) }
+	if m1 > Int64.max/m2 { return Int64.max }
+	if m1 < Int64.min/m2 { return Int64.min }
+	return m1 * m2;
+}
+
+// Returns m1 * m2, clamped to the range [Int64.min, Int64.max].
+// Because of the way this function is used, we can always assume
+// that m2 > 0.
+private func clampedInt64Product(_ m1: Double, _ m2: Double) -> Int64 {
```

Comment:
…and I'm not sure why this isn't just a transformation on a single Double.

Reply:
Can you explain what you mean by that?

Reply:
It looks like it's equivalent to `toInt64Clamped(m1 * m2)`.